### PR TITLE
Toaster loader common header

### PR DESCRIPTION
**Pull request by uibhanu5**

### CODE REVIEW

File: src/app/shared/services/common/loader/loader.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { SPINNER_TYPE } from './spinner-enums';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoaderService {
+  constructor(public loadingController: LoadingController) {}
+
+  async showLoader(
+    messageValue = 'Loading, Please Wait...',
```
Comment (Nithyam1330) [inline src/app/shared/services/common/loader/loader.service.ts:25]: Can you make an interface for this and export it ?

File: src/app/shared/services/common/toaster/toaster.service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { ToastController } from '@ionic/angular';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ToasterService {
+  constructor(private _toastController: ToastController) { }
+
+  async presentToast(message, color) {
+    const toast = await this._toastController.create({
```
Comment (Nithyam1330) [inline src/app/shared/services/common/toaster/toaster.service.ts:21]: Can you make a Interface here also

File: src/app/tab3/login/login.component.ts [hunk 3]
```diff
@@ -7,8 +10,34 @@ import { Component, OnInit } from '@angular/core';
...
+    private commonRequestService: CommonRequestService) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.initLoginForm();
+  }
 
+  initLoginForm() {
+    this.loginForm = this.formBuilder.group({
+      username: ['', Validators.compose([Validators.required, Validators.pattern("[a-zA-Z0-9.-_]{1,}@[a-zA-Z.-]{2,}[.]{1}[a-zA-Z]{2,}")])],
```
Comment (Nithyam1330) [inline src/app/tab3/login/login.component.ts]: Move this Validations into Validation Patterns.ts file and use from there.

File: src/app/tab3/register/register.component.ts [hunk 4]
```diff
@@ -1,14 +1,53 @@
...
-
-  ngOnInit() {}
-
+  ngOnInit() {
+    this.initRegistration();
+  }
+  initRegistration() {
+    this.registerForm = this.formBuilder.group({
+      username: ['', [Validators.required, Validators.pattern("[a-zA-Z0-9.-_]{1,}@[a-zA-Z.-]{2,}[.]{1}[a-zA-Z]{2,}")]],
+      provider:['EMAIL'],
```
Comment (Nithyam1330) [inline src/app/tab3/register/register.component.ts]: Use this Hard coded values from the app-properties.enum file if exist else create a file inside shared/enums

File: src/app/tab3/register/register.component.ts [hunk 5]
```diff
@@ -1,14 +1,53 @@
...
+      role: ['USER'],
+    });
+  }
+  onSubmit() {
+    console.log("hello");
+    const username = this.registerForm.get('username').value;
+   // const provider="EMAIL";
```
Comment (Nithyam1330) [inline src/app/tab3/register/register.component.ts]: Remove this Unwanted code

File: src/app/tab3/tab3.module.ts [hunk 6]
```diff
@@ -1,26 +1,28 @@
...
 import { ResetPasswordComponent } from './reset-password/reset-password.component';
+import { SharedModule } from '../shared/shared.module';
 
 @NgModule({
   imports: [
     IonicModule,
-    CommonModule,
-    FormsModule,
+    CommonModule,SharedModule,
+    FormsModule,ReactiveFormsModule,
```
Comment (Nithyam1330) [inline src/app/tab3/tab3.module.ts]: Add this Forms and Reactive Forms module in sharedModule and use by Shared Module

File: src/app/tab3/tab3.page.ts [hunk 7]
```diff
@@ -8,33 +10,44 @@ import { ResetPasswordComponent } from './reset-password/reset-password.componen
...
   templateUrl: 'tab3.page.html',
-  styleUrls: ['tab3.page.scss']
+  styleUrls: ['tab3.page.scss'],
 })
 export class Tab3Page {
-
-  constructor(private modalController: ModalController) {}
+  constructor(
+    private modalController: ModalController,
+    private _loader: LoaderService,
```
Comment (Nithyam1330) [inline src/app/tab3/tab3.page.ts]: Remove underscores. This is deprecated in the new esLint Rules

File: src/app/tab3/register/register.component.ts (hunk 4 above)
Comment (Nithyam1330) [reply src/app/tab3/register/register.component.ts]: User validation.compose.

File: src/app/shared/services/common/loader/loader.service.ts (hunk 1 above)
Comment (uibhanu5) [reply src/app/shared/services/common/loader/loader.service.ts:25]: Done

File: src/app/shared/services/common/toaster/toaster.service.ts (hunk 2 above)
Comment (uibhanu5) [reply src/app/shared/services/common/toaster/toaster.service.ts:21]: Done

File: src/app/tab3/login/login.component.ts (hunk 3 above)
Comment (uibhanu5) [reply src/app/tab3/login/login.component.ts]: Done

File: src/app/tab3/register/register.component.ts (hunk 4 above)
Comment (uibhanu5) [reply src/app/tab3/register/register.component.ts]: Done

File: src/app/tab3/register/register.component.ts (hunk 5 above)
Comment (uibhanu5) [reply src/app/tab3/register/register.component.ts]: Done

File: src/app/tab3/tab3.module.ts (hunk 6 above)
Comment (uibhanu5) [reply src/app/tab3/tab3.module.ts]: Done

File: src/app/tab3/tab3.page.ts (hunk 7 above)
Comment (uibhanu5) [reply src/app/tab3/tab3.page.ts]: Done

File: src/app/shared/constants/registration-constants.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+export abstract class ProviderTypes {
```
Comment (Nithyam1330) [inline src/app/shared/constants/registration-constants.ts:1]: Nice to See this

File: src/app/tab3/forgot-password/forgot-password.component.html [hunk 9]
```diff
@@ -21,11 +21,15 @@
...
       <ion-card-content>
         <form [formGroup]="ForgetPasswordForm" (ngSubmit)="onSubmit()">
-        <div class="h-px-75">
+        <div class="h-px-90">
           <ion-item>
             <ion-label position="floating">Email</ion-label>
             <ion-input formControlName="username" type="text" placeholder="Enter email"></ion-input>
           </ion-item>
+          <ion-text color="danger" class="f14"
+          *ngIf="ForgetPasswordForm.get('username').invalid && ForgetPasswordForm.get('username').touched">
```
Comment (Nithyam1330) [inline src/app/tab3/forgot-password/forgot-password.component.html:30]: No need to do all this we have a validity method in the base class and that will take care of this. Please use that Utility.

**Pull request merged**